packages/components/src/Combobox/hooks/useComboboxContent.ts

```diff
@@ -1,38 +1,57 @@
 import { useEffect, useRef, useState } from "react";
 import { ComboboxOption } from "../Combobox.types";
 
-export function useComboboxContent(
-  options: ComboboxOption[],
-  open: boolean,
-): {
+interface useComboboxContent {
   searchValue: string;
   setSearchValue: React.Dispatch<React.SetStateAction<string>>;
-  setSelectedElement: React.Dispatch<React.SetStateAction<HTMLElement | null>>;
+  setFirstSelectedElement: React.Dispatch<
+    React.SetStateAction<HTMLElement | null>
+  >;
   filteredOptions: ComboboxOption[];
   optionsListRef: React.RefObject<HTMLUListElement>;
-} {
+  selectedOptions: ComboboxOption[];
+  setInternalSelected: (selected: ComboboxOption[]) => void;
+}
+
+export function useComboboxContent(
+  options: ComboboxOption[],
+  open: boolean,
+  selected: ComboboxOption[],
+  onClose: ((selection: ComboboxOption[]) => void) | undefined,
+): useComboboxContent {
   const [searchValue, setSearchValue] = useState<string>("");
-  const [selectedElement, setSelectedElement] = useState<HTMLElement | null>(
-    null,
-  );
+  const [firstSelectedElement, setFirstSelectedElement] =
+    useState<HTMLElement | null>(null);
   const optionsListRef = useRef<HTMLUListElement>(null);
   const filteredOptions = options.filter(option =>
     option.label.toLowerCase().includes(searchValue.toLowerCase()),
   );
 
+  const [internalSelected, setInternalSelected] =
+    useState<ComboboxOption[]>(selected);
+  const selectedOptions = onClose ? internalSelected : selected;
+
+  useEffect(() => {
+    if (!open && onClose) {
+      onClose(selectedOptions);
+    }
+  }, [open, onClose]);
```

Comment:
this feels wrong to omit `selectedOptions` but this works fine.... why does that work though?

oh it's because of the `useState` of `internalSelected` that is getting modified and updating this hook. I should probably just add to the dependencies. if we ever moved that `useState` out of here I believe this would stop working, plus someday when we add the exhaustive dependencies eslint rule it's going to yell at us for not having it